### PR TITLE
WebAPI: Get database endpoint

### PR DESCRIPTION
**Pull request by marcoandredinis**
This commit creates a new endpoint to return a single database by its name.

```
GET /webapi/sites/:site/databases/:database

{
  "name": "<database_name>",
  "desc": "<description>",
  "protocol": "<protocol>",
  "type": "<type>",
  "labels": [
    {"name": "<name>", "value": "<value>"},
    ...
  ],
  "database_users": ["<db_user1>", ...],
  "database_names": ["<db_name1>", ...],
}
```

The `database_users` and `database_names` will list the available Database User and Names for the current user, based on its roles.

### CODE REVIEW

**Comment by greedy52** [comment]
Should we get `types.Database` or `types.DatabaseServer`?

**Comment by marcoandredinis** [comment]
> Should we get `types.Database` or `types.DatabaseServer`?

Hey :wave: 
Thank you for looking into this.

The goal is to expose the available "static" Database Users and Database Names when showing this screen
In this context, static is the list of DB Users and Names coming from the user roles.
image

I think using `Database` instead of `DatabaseServer` is the correct approach here.

**Review by mdwn — APPROVED** [review]
LGTM.

**Comment by greedy52** [comment]
> The goal is to expose the available "static" Database Users and Database Names when showing this screen In this context, static is the list of DB Users and Names coming from the user roles.
Ah, I see. Thanks for providing context. For db users and names, do you need to fetch the database resource?

**Comment by marcoandredinis** [comment]
> Ah, I see. Thanks for providing context. For db users and names, do we need to fetch the database resource?

I applied the same logic we have for [Server SSH logins](https://github.com/gravitational/teleport/blob/564195819444164eb0091ef60a382fd5413a1ef0/lib/web/ui/server.go#L53) and [Kube Users and Groups](https://github.com/gravitational/teleport/blob/564195819444164eb0091ef60a382fd5413a1ef0/lib/web/ui/server.go#L115-L119)
When querying a resource, we also return the available logins (ssh logins, kube user/groups, database user/names) as part of the response using the current user's roles

**Comment by marcoandredinis** [comment]
> Should we get `types.Database` or `types.DatabaseServer`?

I'm pretty sure we will create a `GET .../database_server/:name` endpoint as well

We'll need to get the DatabaseServer so that we know when the database service/agent proxy is up
I don't know the details yet, but I think we'll need it

File: lib/web/apiserver_test.go [hunk 1]
```diff
@@ -2709,6 +2710,136 @@ func TestClusterDatabasesGet(t *testing.T) {
...
 }
 
+func TestClusterDatabaseGet(t *testing.T) {
+	env := newWebPack(t, 1)
+	ctx := context.Background()
+
+	proxy := env.proxies[0]
+
+	// Register a database.
+	dbName := "existingdb"
```
Comment (jakule) [inline lib/web/apiserver_test.go:2720]: nit:
```suggestion
	const dbName = "existingdb"
```

File: lib/web/apiserver_test.go [hunk 2]
```diff
@@ -2709,6 +2710,136 @@ func TestClusterDatabasesGet(t *testing.T) {
...
+				require.NoError(tt, err)
+
+				return []types.Role{roleWithDBUser, roleWithDBName}
+			},
+			expectedDBNames: dbNames,
+			expectedDBUsers: dbUsers,
+			requireError:    require.NoError,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (jakule) [inline lib/web/apiserver_test.go:2816]: Will this test work if we add `t.Parallel()` here?

File: lib/web/servers.go [hunk 3]
```diff
@@ -85,6 +85,46 @@ func (h *Handler) clusterDatabasesGet(w http.ResponseWriter, r *http.Request, p
 	}, nil
 }
 
+// clusterDatabaseGet returns a list of db servers in a form the UI can present.
+func (h *Handler) clusterDatabaseGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (interface{}, error) {
```
Comment (jakule) [inline lib/web/servers.go:89]: nit:
```suggestion
func (h *Handler) clusterDatabaseGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, ctx *SessionContext, site reversetunnel.RemoteSite) (any, error) {
```

File: lib/web/apiserver_test.go [hunk 4]
```diff
@@ -2709,6 +2710,136 @@ func TestClusterDatabasesGet(t *testing.T) {
...
+				)
+				require.NoError(tt, err)
+				return []types.Role{role}
+			},
+			requireError: func(tt require.TestingT, err error, i ...interface{}) {
+				require.True(tt, trace.IsAccessDenied(err), "expected a access denied error, got %v", err)
+			},
+		},
+		{
+			name:         "authorizedDBNamesUsers",
```
Comment (jakule) [inline lib/web/apiserver_test.go:2780]: Can you also add a test case with no databases? This can help with some corner cases.

**Review by jakule — APPROVED** [review]

File: lib/web/servers.go [hunk 5]
```diff
@@ -85,6 +85,46 @@ func (h *Handler) clusterDatabasesGet(w http.ResponseWriter, r *http.Request, p
...
+	if databaseName == "" {
+		return nil, trace.BadParameter("database name is required")
+	}
+
+	clt, err := ctx.GetUserClient(site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	db, err := clt.GetDatabase(r.Context(), databaseName)
```
Comment (r0mant) [inline lib/web/servers.go:100]: As discussed, we should be fetching database servers here.

Comment (marcoandredinis) [reply lib/web/servers.go:100]: Changing this to DatabaseServers

cc @greedy52

**Review by greedy52 — APPROVED** [review]

File: lib/web/servers.go (hunk 5 above)
Comment (marcoandredinis) [reply lib/web/servers.go]: @r0mant Can you please take another look?
I'm now using DatabaseServers as source

**Review by r0mant — APPROVED** [review]
